### PR TITLE
tool for creating atlases via virtual objects

### PR DESCRIPTION
**Pull request by drh-stanford**
A tool to merge multiple objects into a parent object via virtual object structure, namely the externalFile element in contentMetadata.

### CODE REVIEW

File: bin/virtual-merge [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        @logger.error "#{e}: #{e.backtrace.join("\n")}"
+      end
+    end
+  end
+end
+
+# __MAIN__
+
+# load Dor configuration
+require_relative '../config/environments/development'
```
Comment (atz) [inline bin/virtual-merge]: Why always load the development environment?

Comment (drh-stanford) [inline bin/virtual-merge]: Thanks. It shouldn't. I've pushed a fix to the PR

**Comment by drh-stanford** [comment]
@atz please take a look at this virtual atlas creation tool. it requires code from https://github.com/sul-dlss/dor-services/pull/79 on the 5.x branch to run.

**Comment by atz** [comment]
So to make this production ready, the #79 PR needs to be merged and released.  Then this PR should bump the dor-services requirement.  Is that right?  But is this repo OK with running against 5.x when the data hasn't been fully converted/indexed yet?

**Comment by drh-stanford** [comment]
Thanks Joe. I wasn't aware of the 5.x data requirements. I'll switch https://github.com/sul-dlss/dor-services/pull/79 to the 4.x branch. And make this PR use the upcoming dor-services 4.x release.

**Comment by drh-stanford** [comment]
PR sul-dlss/dor-services#91 is now the 4.x dor-services code that this utility needs to add virtual resources -- namely `ContentMetadataDS.add_virtual_resource` and `Contentable.add_constituent`.

**Comment by drh-stanford** [comment]
I've added code for JUMBO-29 (the `dor-services' allow_modification?` calls) and added some brief documentation on usage to the README

**Comment by drh-stanford** [comment]
This PR is blocked on getting a new `dor-services` 4.x release that includes https://github.com/sul-dlss/dor-services/pull/89, https://github.com/sul-dlss/dor-services/pull/91, https://github.com/sul-dlss/dor-services/pull/92

**Comment by drh-stanford** [comment]
Also blocked on https://github.com/sul-dlss/digital_stacks_rails/issues/18

**Comment by drh-stanford** [comment]
I've updated dor-services to 4.23.0 which has all the code that the `virtual-merge` tool needs. The PR is ready for review.

**Comment by atz** [comment]
Might be some sense to breaking the class out to its own file for reuse and testing, but I'm happy to defer that until whatever time we actually need it.

**Pull request merged**